### PR TITLE
Fix clippy lints

### PR DESCRIPTION
**Pull request by magnusuMET**

### CODE REVIEW

File: netcdf/src/attribute.rs [hunk 1]
```diff
@@ -104,6 +104,7 @@ impl<'a> Attribute<'a> {
                     Ok(AttrValue::Uchar(value))
                 }
                 len => {
+                    #[allow(clippy::unnecessary_cast)]
```
Comment (lnicola) [inline netcdf/src/attribute.rs:107]: Is it worth keeping the cast here and in `File::path`?

Comment (magnusuMET) [reply netcdf/src/attribute.rs:107]: This one is indeed a NOP, but the one in `File::path` changes from `size_t` to `usize` which could be different

Comment (lnicola) [reply netcdf/src/attribute.rs]: Right, clippy flags it because of https://github.com/georust/netcdf/blob/master/netcdf-sys/src/functions.rs#L52.

Comment (magnusuMET) [reply netcdf/src/attribute.rs]: Here are the docs for `size_t`: https://doc.rust-lang.org/core/ffi/type.c_size_t.html. I will remove the cast, thanks!

File: netcdf/src/error.rs [hunk 2]
```diff
@@ -129,7 +129,7 @@ use std::fmt;
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Self::Str(x) => write!(f, "{}", x),
+            Self::Str(x) => write!(f, "{x}"),
```
Comment (lnicola) [inline netcdf/src/error.rs:132]:
```suggestion
            Self::Str(x) => x.fmt(f),
```

File: netcdf/src/error.rs [hunk 3]
```diff
@@ -145,9 +145,9 @@ impl fmt::Display for Error {
                 "buffer size mismatch, has size {actual}, but needs size {wanted}",
             ),
             Self::TypeMismatch => write!(f, "netcdf types does not correspond to what is defined"),
-            Self::TypeUnknown(t) => write!(f, "netcdf type {} is not known", t),
+            Self::TypeUnknown(t) => write!(f, "netcdf type {t} is not known"),
             Self::AlreadyExists => write!(f, "variable/group/dimension already exists"),
-            Self::NotFound(x) => write!(f, "Could not find {}", x),
+            Self::NotFound(x) => write!(f, "Could not find {x}"),
```
Comment (lnicola) [inline netcdf/src/error.rs:150]: These are inconsistent, but I think most errors will display themselves with a lower-case letter.
```suggestion
            Self::NotFound(x) => write!(f, "could not find {x}"),
```

File: netcdf/src/error.rs [hunk 4]
```diff
@@ -162,8 +162,8 @@ impl fmt::Display for Error {
             Self::Overflow => write!(f, "slice would exceed maximum size of possible buffers"),
             Self::Conversion(e) => e.fmt(f),
             Self::WrongDataset => write!(f, "This identifier does not belong in this dataset"),
-            Self::Utf8Conversion(e) => write!(f, "{}", e),
-            Self::NulError(e) => write!(f, "String value contains null bytes {}", e),
+            Self::Utf8Conversion(e) => write!(f, "{e}"),
```
Comment (lnicola) [inline netcdf/src/error.rs:165]:
```suggestion
            Self::Utf8Conversion(e) => e.fmt(f),
```

**Review by lnicola — APPROVED** [review]

File: netcdf/src/error.rs (hunk 2 above)
Comment (magnusuMET) [reply netcdf/src/error.rs:132]: I find this awkward and inconsistent with the other entries, is there a benefit to this in terms of code generation?

Comment (lnicola) [reply netcdf/src/error.rs:132]: Yeah, it produces much less code, at least until someone changes it back to add an error message 😅

Comment (magnusuMET) [reply netcdf/src/error.rs:132]: Makes sense, I changed it to match your suggestions

**Comment by lnicola** [comment]
For the record, https://rust.godbolt.org/z/jYxeM74nr. This part is the code for `Error`:

image

`Error2` has a similar snippet:

image

But also a whole bunch of extra code:

image

**Pull request merged**